### ISSUE TITLE
Wahoo Elemnt Bolt Libre2 support

### ISSUE THREAD

**Issue by maartenbrakkee** [NightscoutFoundation/xDrip#1654]
### Is your feature request related to a problem? Please describe.
No.

### Describe the solution you'd like
With a patched LibreLink app, sensor data is sent to xDrip and xDrip communicates it towards the Wahoo Elemnt Bolt. Because the phone is already connected to the Wahoo Elemnt Bolt, setting the device as a collection device would not be necessary (and both the phone and the original reader could keep their connection to the sensor).

### Describe alternatives you've considered
I thought of following https://github.com/NightscoutFoundation/xDrip/wiki/How-to-use-the-Wahoo-Elemnt-Bolt-cycle-computer-as-a-collection-device, but is was already found not working https://github.com/NightscoutFoundation/xDrip/issues/1500.

**Comment by tolot27** [comment]
What is your feature request, exactly? With Libre2 it is impossible to configure more than one device as a collector.

**Comment by maartenbrakkee** [comment]
Thank you for your reply. I assumed Libre2 worked like Libre1 (first start the sensor with the original libre reader, after you can also add your phone).

Given that you've only linked your Libre2 to your phone (with patched Librelink connected to xDrip), it would be nice to see my levels on the Wahoo Elemnt Bolt. Displaying the levels in the same way as when the Wahoo is being used as a collector for a G6 (as seen in this video: https://www.youtube.com/watch?v=1VoZAgXCQcM).

In this case, you would have your phone with you and therefore it would not be needed for the Wahoo to act as a collector. There is already a connection between your phone and the Wahoo, maybe xDrip can use this to send the information to the Wahoo?

**Comment by Navid200** [comment]
@maartenbrakkee Would you please close this issue if it has been resolved?
Please help us by doing a little bit of research, using the support channels, before opening an issue.
https://www.facebook.com/groups/xDripG5
https://gitter.im/jamorham/xDrip-plus
Thanks

**Comment by maartenbrakkee** [comment]
@Navid200 the issue is not resolved. I've searched both https://gitter.im/jamorham/xDrip-plus and https://github.com/NightscoutFoundation/xDrip/issues/ before opening this issue/feature request.

If there is anything unclear about the feature request, please let me know.

**Comment by derboo1** [comment]
I`m also very interested in this feature. Would love to see this

**Comment by scammels** [comment]
> @Navid200 the issue is not resolved. I've searched both https://gitter.im/jamorham/xDrip-plus and https://github.com/NightscoutFoundation/xDrip/issues/ before opening this issue/feature request.
> 
> If there is anything unclear about the feature request, please let me know.

im also interested.

**Comment by tvt92** [comment]
I also only have the scanning field on my Wahoo element and would like to see my values. I searched all support pages intensively and found nothing. If someone has a solution, I would be grateful.

**Comment by fatterthantheshadow** [comment]
I too am interested in this, I own a Wahoo Element bolt, I have NHS funding for Libre 1 but don't have access to Dexcom products. 
My AAPS modules are L1-Miao2- xdrip+ - Samsung M30 phone - AAPS- accucheck Insight - Nightscout - HEROKU- android wear watch with AAPS app.
Just like maartenbrakkee suggests, is it not possible to connect the Wahoo to some place within this set up, as loopers we always have the phone with us, just not visible when riding a bike. I believe Wahoo runs on an Android os, is there no way to push the cgm data to the wahoo directly from xdrip, Nightscout or AAPS on the phone rather than it being a collector?

**Comment by psonnera** [comment]
Did you see this? https://github.com/NightscoutFoundation/xDrip/wiki/How-to-use-the-Wahoo-Elemnt-Bolt-cycle-computer-as-a-collection-device

**Comment by tvt92** [comment]
@psonnera: yes, but everything that can be found on the net only works with Dexcom G5/G6 and not with Libre 1 or 2.

**Comment by flashydog** [comment]
Also very interested in this.  Just got the Libre2 and managed to get it to display my Garmin fenix5 which is amazing. 

If I could also get the readings to display on the Wahoo Element Bolt I wont have to buy a Garmin Edge.  

Would be amazing to have this on handlebars when out on the MTB and see at a glance if in range.

**Comment by stev0n** [comment]
I would like this as well.

**Comment by tonydeleon9** [comment]
I would also love this feature too please

**Comment by flejak** [comment]
I would love this feature too!

**Comment by nahuis** [comment]
I invented a workaround. Not as a data field, but via email messages. You should have set up Nightscout. Works with any xDrip, regardless which device. 
https://www.facebook.com/groups/cgminthecloud/permalink/4101077896680102/

**Comment by Navid200** [comment]
We shouldn't have multiple open issues related to the same problem.
https://github.com/NightscoutFoundation/xDrip/issues/1500
It only clutters the issues platform, which diminishes the significance of all open issues.

**Comment by nahuis** [comment]
This is a different and unsolved problem. Please remove the duplicate label.

There is no solution yet for remotely following loved ones via xDrip on the Wahoo. Neither for getting bg values from sensors other than G6. If I had the skills, I would immediately solve it. It should be easy.

People try to use the G6 collector patch for this problem, but that doesn't work. The issue you refer to is about removing the G6 data field from a Wahoo, after discovering that it doesn't work with Libre.

**Comment by Navid200** [comment]
Can you explain in more detail?  Where should we draw the line?
Both issues are requesting for xDrip to support Wahoo.  Do you think that if a developer spends time to solve one, they cannot address both?
I have to take your word for the two being completely different.  can you help me understand how different they are?

As an example, please let me use an analogy.  Let's say someone opens a request for xDrip to support a new Device (transmitter).  It could be Dexcom G7.
Someone else opens another request for xDrip to allow xDrip to calibrate G7.

I would consider those two requests to be dependent on each other.  There is no way for xDrip to calibrate a device that it cannot connect to.  So, the first request needs to be addressed before the second one can be considered.    But, I wouldn't have those two requests open.  I would add a comment to the first request that after support is provided we will need to be able to calibrate it as well.  Now, if some day support is added but we cannot calibrate, then, it would make sense to open a new request.

Can you tell me why these two requests are completely independent?
This is not a hypothetical question.  I can really use your help understanding.

**Comment by nahuis** [comment]
Issue #1500 can be closed, then this issue is not a duplicate anymore.

For your understanding, the difference is between Wahoo support for G6 and Wahoo support for xDrip.

**Comment by Navid200** [comment]
You are not the owner of either of the two issues.  
We usually maintain the older issue in order to preserve the date the issue was originally raised.

**Comment by tolot27** [comment]
If I understand the issue correctly, it is requested that the Wahoo Elemnt devices (i. e. Roam or Bolt) connects to xDrip to show the BG data which is collected by xDrip on the phone (regardless of the sensor source). It should act like some Android Smart Watches work.
If this understanding is correct, the title should be changed.

I thought this works already by simply enabling "Android Wear Integration" in the "Smart Watch Features" settings.

**Comment by nahuis** [comment]
Wahoo does not facilitate third party data fields. Yet, @jamorham managed to add a bg value data field to the Wahoo, with bg directly taken from the g6 (https://github.com/NightscoutFoundation/xDrip/wiki/How-to-use-the-Wahoo-Elemnt-Bolt-cycle-computer-as-a-collection-device) 

The request here is to add a similar bg data field to the Wahoo, but with bg values taken from xDrip.

**Comment by tolot27** [comment]
Do you know when exactly this feature was added? I could not find anything in the source files about Wahoo, Elemnt, Bolt or .fit files. As long as I don't know how @jamorham implemented the display of the bg value, I cannot estimate if it would work with xDrip as a data source.

**Comment by nahuis** [comment]
Sorry, I don't know. Maybe March or April 2019.

**Comment by tolot27** [comment]
@jamorham When did you implement the collector for Wahoo Elemnt? Please can you link the corresponding code here?

**Comment by FamouZzE** [comment]
Hello, on my search to add my blood sugar to my Wahoo Element Roam, I came across this thread. Since there was no change in the last two years I hope to revive it and if I am able to maybe help implement this feature. So I started some digging for the code for dexcom and think I found the following Info:
[Here](https://github.com/NightscoutFoundation/xDrip/blob/f9346abc5870f6b7a86a56140a2025c5521073f8/app/src/main/java/com/eveningoutpost/dexdrip/utilitymodels/MtpConfigure.java#L119) you can see that a file is written to the Wahoo called xdrip-install.dat if I follow the getBytes() Call from that method I end up [here](https://github.com/NightscoutFoundation/xDrip/blob/f9346abc5870f6b7a86a56140a2025c5521073f8/app/src/main/java/com/eveningoutpost/dexdrip/utils/GetWearApk.java#L87) which for me sounds like the xdrip wear apk is written and probably installed to the Wahoo device.
Maybe someone can take a look at it or point me in the right direction because I think this would be a very neat feature which would enable users to have any Source that can be connected to Nightscout on their Wahoo Unit.

**Comment by CadzyT** [comment]
The new Wahoo Roam v2 works with Supersapien. Does anyone know about Supersapien? I understand it is not suitable for type 1 diabetic monitoring.

**Comment by psonnera** [comment]
Supersapiens sensors are not for type 1 diabetes and are not supported by xDrip+ AFAIK.

**Comment by gui-dos** [comment]
The Libre Sense “biosensor” is just a rebrand of the Libre 2 US/CA/AU; they differ only for their regional code: `01`: Europe, `02` :US,  `04`: CA/AU, otherwise the "Gen2" protocols they adopt are the same.

The upper 200 mg/dL probably is just a software limitation.

https://tryterra.co/podcast?episode=Founder-and-CEO-at-Supersapiens-Phil-Southerland

**Comment by CadzyT** [comment]
Yes, that's what I thought. It's a shame it's set up only for those who do not need to monitor their glucose but not for type 1 diabetics

**Comment by bobcosine** [comment]
I have been using G6 and Wahoo Elemnt Bolt for a couple of years.  Now I have G7.  Tried the usual way to connect to the Bolt but it will not work.  Bolt reboots but the BG field is gone.  Tried about 5 times.  I do have the G7 readings in xdrip+ which was not easy but would really like to get it on the Bolt.  Any thoughts appreciated.
